### ISSUE TITLE
keras prediction always give value 0

### ISSUE THREAD

**Issue by TonyFly3000** [rstudio/keras#915]
I am using both R and python to predict a same photo with same model .
But in R , the prediction result always 0 or 1 .where in python will give me a correct probability .

 --------------   R keras ----------------------------

img <- image_load("dog_sample001.jpg", target_size = c(150,150))
x <- image_to_array(img)
x <- array_reshape(x, c(1, dim(x)))
x <- imagenet_preprocess_input(x)

new_model <- load_model_hdf5("dog_cat_cnn_model.h5")
result <- dog_cat_cnn_model_new %>% predict(x)
result

> result
     [,1]
[1,]    0

 --------------   Python keras ----------------------------
import numpy as np
import PIL.Image as Image
import tensorflow as tf

x = Image.open('dog_sample001.jpg').resize((150, 150))
x = np.array(x)/255.0
new_model = tf.keras.models.load_model ('dog_cat_cnn_model.h5')
result = new_model.predict(x[np.newaxis, ...])
result

result
Out[17]: array([[0.9128384]], dtype=float32)
-----------------------------------------------------

**Comment by skeydan** [comment]
Can you try what happens if you use the same preprocessing in both cases?
(Either `imagenet_preprocess_input` on both, or `/255` on both).

Default mode `caffe` does not seem to scale the inputs (https://github.com/keras-team/keras-applications/blob/master/keras_applications/imagenet_utils.py)

```
if mode == 'tf':
        x /= 127.5
        x -= 1.
        return x
if mode == 'torch':
        x /= 255.
        mean = [0.485, 0.456, 0.406]
        std = [0.229, 0.224, 0.225]
```

**Comment by TonyFly3000** [comment]
Thanks. After I change the R keras code as following . I am able to get the same result as python keras 

-------------- R keras ----------------------------

img <- image_load("dog_sample001.jpg", target_size = c(150,150))
x <- image_to_array(img)
x <- array_reshape(x, c(1, dim(x)))/255.0


new_model <- load_model_hdf5("dog_cat_cnn_model.h5")
result <- dog_cat_cnn_model_new %>% predict(x)
result
